.changeset/plain-mails-tan.md

```diff
@@ -0,0 +1,24 @@
+---
+"@fluidframework/aqueduct": minor
+"@fluidframework/container-runtime": minor
+---
+---
+"section": deprecation
+---
+
+The ContainerRuntime class is now deprecated
+
+The class `ContainerRuntime` is deprecated and will no longer be exported starting in version 2.20.0.
+
+There are two possible migration paths to stop using `ContainerRuntime`:
+
+* When using it as a type, replace it with an interface like `IContainerRuntime` (or in rare cases, `IRuntime`)
+* When using the static function `ContainerRuntime.loadRuntime` replace it with the free function `loadContainerRuntime`.
+
+`BaseContainerRuntimeFactory` has some changes as well, since it exposed `ContainerRuntime` in several function signatures:
+
+* `instantiateFirstTime` - Takes the wider type `IContainerRuntime` instead of `ContainerRuntime`
+* `instantiateFromExisting` - Takes the wider type `IContainerRuntime` instead of `ContainerRuntime`
+* `preInitialize` - deprecated as well, since it returns `ContainerRuntime`
```

Comment:
`BaseContainerRuntimeFactory` is not `@sealed` so right now people could in theory subclass it, and I believe that would make this breaking for anyone that was doing it. I don't know how confident we are that it is not happening today... and if we are confident enough, maybe we should mark it `@sealed`. And well, depending on that level of confidence, these might actually need to wait until 2.20?

Reply:
Great call-out.  I discussed this at length with @jason-ha, it's technically not a breaking change based on TypeScript weirdness (kind of disappointing), but you're right that it would lead to unexpected behavior if a subclass was counting on getting `ContainerRuntime` and then a caller provided something outside that.

HOWEVER -- in the ecosystem in which this is used, it'll always be `ContainerRuntime`.  We just don't want to talk about it :)